src/Mexitek/PHPColors/Color.php

```diff
@@ -196,9 +196,33 @@ public static function rgbToHex( $rgb = array() ){
         $hex[1] = str_pad(dechex($rgb['G']), 2, '0', STR_PAD_LEFT);
         $hex[2] = str_pad(dechex($rgb['B']), 2, '0', STR_PAD_LEFT);
 
+        // Make sure that 2 digits are allocated to each color.
+        $hex[0] = (strlen($hex[0]) == 1)? '0'.$hex[0] : $hex[0];
+        $hex[1] = (strlen($hex[1]) == 1)? '0'.$hex[1] : $hex[1];
+        $hex[2] = (strlen($hex[2]) == 1)? '0'.$hex[2] : $hex[2];
```

Comment:
These lines are the core of this bugfix.

Other changes implement and support tests, or cleanup.